src/test/java/api/CirculationRulesAPITests.java

```diff
@@ -5,31 +5,138 @@
+
   @Test
   public void canGet() {
     getRulesText();
   }
 
   @Test
   public void canPutAndGet() {
-    String rule = "priority: t, s, c, b, a, m, g\nfallback-policy: l no-circulation r no-hold n basic-notice o basic-overdue i basic-lost-item\n";
+    UUID lp1 = UUID.randomUUID();
+    UUID lp2 = UUID.randomUUID();
+    UUID rp1 = UUID.randomUUID();
+    UUID rp2 = UUID.randomUUID();
+    UUID np1 = UUID.randomUUID();
+    UUID np2 = UUID.randomUUID();
+    UUID op1 = UUID.randomUUID();
+    UUID op2 = UUID.randomUUID();
+    UUID ip1 = UUID.randomUUID();
+    UUID ip2 = UUID.randomUUID();
+
+    loanPoliciesFixture.create(lp1);
+    noticePoliciesFixture.create(np1);
+    requestPoliciesFixture.allowAllRequestPolicy(rp1);
+    overdueFinePoliciesFixture.create(op1);
+    lostItemFeePoliciesFixture.create(ip1);
+    loanPoliciesFixture.create(lp2);
+    noticePoliciesFixture.create(np2);
+    requestPoliciesFixture.allowAllRequestPolicy(rp2);
+    overdueFinePoliciesFixture.create(op2);
+    lostItemFeePoliciesFixture.create(ip2);
+
+    String rule = String.format(CIRCULATION_RULE_TEMPLATE, lp1, rp1, np1, op1, ip1);
+    setRules(rule);
+
+    assertThat(getRulesText(), is(rule));
 
-    circulationRulesFixture.updateCirculationRules(rule);
+    rule = String.format(CIRCULATION_RULE_TEMPLATE, lp2, rp2, np2, op2, ip2);
+    setRules(rule);
 
     assertThat(getRulesText(), is(rule));
+  }
 
-    rule = "priority: t, s, c, b, a, m, g\nfallback-policy: l loan-forever r two-week-hold n two-week-notice o forever-overdue i forever-lost-item\n";
+  @Test
+  public void cannotUpdateCirculationRulesWithInvalidLoanPolicyId() {
+
+    String rule = circulationRulesFixture.soleFallbackPolicyRule(
+      UUID.randomUUID().toString(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId().toString(),
+      noticePoliciesFixture.activeNotice().getId().toString(),
+      overdueFinePoliciesFixture.facultyStandard().getId().toString(),
+      lostItemFeePoliciesFixture.facultyStandard().getId().toString());
+
+    Response response = circulationRulesFixture
+      .attemptUpdateCirculationRules(rule, "l");
+    assertThat("The policy l does not exist",
+      response.getStatusCode(), Is.is(422));
+  }
 
-    circulationRulesFixture.updateCirculationRules(rule);
+  @Test
+  public void cannotUpdateCirculationRulesWithInvalidNoticePolicyId() {
+
+    String rule = circulationRulesFixture.soleFallbackPolicyRule(
+      loanPoliciesFixture.canCirculateFixed().getId().toString(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId().toString(),
+      UUID.randomUUID().toString(),
+      overdueFinePoliciesFixture.facultyStandard().getId().toString(),
+      lostItemFeePoliciesFixture.facultyStandard().getId().toString());
+
+    Response response = circulationRulesFixture
+      .attemptUpdateCirculationRules(rule, "n");
+    assertThat("The policy n does not exist",
+      response.getStatusCode(), Is.is(422));
```

Comment:
Can the validation response matchers be used to check more specific aspects of the response?

Something similar to the example below.

```
    assertThat(putResponse.getJson(), hasErrorWith(allOf(
      hasMessage("Check In Service Point does not exist"),
      hasUUIDParameter("checkinServicePointId", unknownServicePointId))));
```

Reply:
We can't check ID here, the response looks like this: {"message":"The policy n does not exist","line":2,"column":15}

Reply:
Why is it not possible to check the ID here?

Reply:
As I see the response doesn't have the ID

Reply:
That assertion was only intended to be an example, if the error response does not include any parameters, that might be ok. If so, please check for the correct message (which I assume contains the relevant ID)

Reply:
Done. 
I've added assertion to check the error message